core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -127,7 +127,7 @@ public void onComplete() {
             shareFetchData.future().complete(result);
         } catch (Exception e) {
             log.error("Error processing delayed share fetch request", e);
-            shareFetchData.future().completeExceptionally(e);
+            sharePartitionManager.handleFetchException(shareFetchData.groupId(), topicPartitionData.keySet(), shareFetchData.future(), e);
```

Comment:
Typically, the error code is returned in responseLogResult. So we need to handle the error there too.

Reply:
As we are going with massive refactor with min bytes PR in delayed share fetch, also the handling of this error will have it's own quite possible cases hence created a jira to take that up in following PR: https://issues.apache.org/jira/browse/KAFKA-17887

Reply:
Yes, it seems this is no longer an issue.

Reply:
Thanks for confirming @junrao, I have closed the ticket.